crates/connectors/ndc-postgres/src/error/convert.rs

```diff
@@ -10,19 +10,22 @@ pub fn execution_error_to_query_error(
     match error {
         Error::Query(query_error) => match &query_error {
             QueryError::VariableNotFound(_) => {
-                connector::QueryError::InvalidRequest(query_error.to_string())
+                connector::QueryError::new_invalid_request(&"Invalid request")
+                    .with_details(serde_json::Value::String(query_error.to_string()))
             }
             QueryError::NotSupported(_) => {
-                connector::QueryError::UnsupportedOperation(query_error.to_string())
+                connector::QueryError::new_unsupported_operation(&"Unsupported operation")
+                    .with_details(serde_json::Value::String(query_error.to_string()))
             }
             QueryError::DBError(_) | QueryError::DBConstraintError(_) => {
-                connector::QueryError::UnprocessableContent(query_error.to_string())
-            }
-            QueryError::MutationConstraintFailed => {
-                connector::QueryError::Other(Box::new(query_error))
+                connector::QueryError::new_unprocessable_content(&"Unprocessable content")
+                    .with_details(serde_json::Value::String(query_error.to_string()))
             }
+            QueryError::MutationConstraintFailed => connector::QueryError::new("Internal error")
+                .with_details(serde_json::Value::String(query_error.to_string())),
         },
-        Error::DB(_) => connector::QueryError::Other(Box::new(error)),
+        Error::DB(_) => connector::QueryError::new("Internal error")
```

Comment:
You could pass in the error itself here, if you'd like to keep the underlying error around

Reply:
If you pass the error itself, eventually the message will be `error.to_string()`

Reply:
All this speaks to a very strange API on the NDC SDK side.

I would like to revise this so that we're not passing around duplicate information.